File: algorithms/167 - Two Sum II - Input array is sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let h = numbers.length - 1,
        l = 0,
        sum;
    while (l < h) {
        sum = numbers[l] + numbers[h];
        if (sum === target) break;
        if (sum > target) {
            h--;
        } else {
            l++;
        }
    }

    return [l + 1, h + 1];
};

File: algorithms/ruby/167 - Two Sum II - Input Array Is Sorted.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
  numbers.each_with_index do |number, index|
    second_value = target - number
    second_index = numbers.bsearch_index { |n| second_value - n }
    return [index + 1, second_index + 1].sort if second_index && second_index != index
  end
end
